master/internal/api_checkpoint.go

```diff
@@ -76,25 +76,38 @@ func (m *Master) canDoActionOnCheckpoint(
 }
 
 func (m *Master) canDoActionOnCheckpointThroughModel(
-	ctx context.Context, curUser model.User, id uuid.UUID,
+	ctx context.Context, curUser model.User, ckptID string,
```

Comment:
could we combine ```canDoActionOnCheckpoint``` and ```canDoActionOnCheckpointThroughModel```? 

Just worried someone could just call ```canDoActionOnCheckpoint``` when they would need to call both

Reply:
my concern is that there might be a scenario where we want to check only for experiment permissions and not model permissions. so this will help separate it out and makes it very explicit that you're checking for both.